packages/babel-parser/src/tokenizer/types.ts

```diff
@@ -351,7 +351,7 @@ export const tt = {
 } as const;
 
 export function tokenIsIdentifier(token: TokenType): boolean {
-  return token >= tt._as && token <= tt.name;
+  return (token >= tt._as && token <= tt.name) || token === tt.placeholder;
```

Comment:
Can we move the placeholder token right after the name token so that we can compare with the placeholder token directly?